### DISCUSSION TITLE
Killing Chrome OS on Chroot Startup

### DISCUSSION THREAD

**Discussion by Dozer257** [dnschneid/crouton#4552] [Q&A] [answered]
When I'm in a chroot, I execute "ps ax -o sess,pgrp,pid,cmd | awk '$4 ~ /\/chrome$/ { print $3 }' | xargs $nodo kill" to kill chrome os processes to free up my memory usage. However, it gets annoying to do it manually everytime so I'm thinking to add it as a start-up command so it will be done automatically. But, I am worried that the command might execute too early so that it kills chrome processes during chroot startup making it corrupt or something. This may sound stupid but I just want to know if the chroot will start just fine if I auto start the chrome processes command because I don't want to lose my chroot lol

Btw I am using Ubuntu Focal Fossa with Xfce.

**Comment by ghost** [comment, marked as answer]
If you wanted to be safe you could create a `.sh` script with the command you use to kill Chrome OS, this could be added to path to make it easy or you could make it run at startup with, let's say, a 10 second wait.

**Reply by Dozer257** [reply]
I don't know how to make a `.sh` script :sweat_smile:

**Reply by ghost** [reply]
1. Use `touch <name>` to create a new file.
 2. Use `nano <name>` or `vi <name>` and type in the commands. 
 3. Use `chmod +x <name>`
 4. Finally add it to your path, or move it to a place like `/bin`, and you can run it by typing the name of your file.

**Reply by Dozer257** [reply]
How do I give it a delay? I'm sorry, I'm I have never made a script so I am a complete noob :sweat_smile:

**Reply by ghost** [reply]
https://stackoverflow.com/questions/32359374/how-could-i-run-a-shell-script-with-delay

**Reply by Dozer257** [reply]
Ok, I got the script working and added it as an autostart application and it's working perfectly, thanks for your help!

**Reply by DennisLfromGA** [reply]
You *could* add the script to your `.bashrc` script so that it's executed whenever you log in and not have to run it by hand afterward.

*Hope this helps,*
-**DennisLfromGA**

**Comment by darkwater4213** [comment]
Pretty sure that's a typo. `awk` throws an error when trying to process `'$4 ~ //chrome$/ { print $3 }'` but succeeds with both `'$4 ~ //chrome$// { print $3 }'` and `'$4 ~ /chrome$/ { print $3 }'`. The second one actually _does_ something -- namely, it kills chrome. Thanks for this, by the way. It's been super helpful :)
I also wrote a tiny script that creates a `kill-chrome` command and puts it in `/bin` as root. In other words, it is, for all intents and purposes, part of the system and can only be executed as root (thanks, sudo) even though it's just a shell script. Executing the below code should do this for you but... no guarantees :)  

```sh  
#!/bin/sh
echo "#\!/bin/sh" | \
tee kill-chrome.sh
echo "ps ax -o sess,pgrp,pid,cmd | \\
awk '\$4 ~ /chrome\$/ { print \$3 }' | \\
xargs \$nodo kill \\
exit" | \
tee -a kill-chrome.sh
mv kill-chrome.sh /bin/kill-chrome
cd /bin
sudo chmod a-w kill-chrome
sudo chmod u+w kill-chrome
sudo chmod +s kill-chrome
sudo chown root:root kill-chrome
exit
```  

There's probably a better way of doing this, but I'm still kind of new to Linux, so this was something of a learning experience for me. If you want to make this a shell script, don't forget to make it executable with `sudo chmod +x install.sh`. I only say this because I did forget... Or used to, anyways.

Hope this helps somewhat.

**Comment by TomTravis** [comment]
what happened to init does it start systemd?
ps ax -o sess,pgrp,pid,cmd |awk '$4 ~ /chrome/ {print $3, $4}'
136 [irq/14-chromeos]
1435 /opt/google/chrome/chrome
1445 /opt/google/chrome/chrome_crashpad_handler
1447 /opt/google/chrome/chrome_crashpad_handler
1453 /opt/google/chrome/chrome
1454 /opt/google/chrome/chrome
1466 /opt/google/chrome/nacl_helper_bootstrap
1471 /opt/google/chrome/chrome
1531 /opt/google/chrome/chrome
1533 /opt/google/chrome/chrome
1605 /opt/google/chrome/chrome
1635 /opt/google/chrome/chrome
4045 /opt/google/chrome/chrome
4100 /opt/google/chrome/chrome
4134 /opt/google/chrome/chrome
4139 /opt/google/chrome/chrome
4245 /opt/google/chrome/chrome
4506 /opt/google/chrome/chrome
5194 /opt/google/chrome/chrome
11277 /opt/google/chrome/chrome
11315 /opt/google/chrome/chrome
13407 /opt/google/chrome/chrome
13507 /opt/google/chrome/chrome
15181 /opt/google/chrome/chrome
17033 /opt/google/chrome/chrome
17539 /opt/google/chrome/chrome
21293 /opt/google/chrome/chrome
24487 /opt/google/chrome/chrome
24652 /opt/google/chrome/chrome

if this what you wish to kill
what is the confusion 

bull)z@localhost:~$ ps ax -o sess,pgrp,pid,cmd |awk '$4 ~ /init/ {print $3, $4, $5}'

1 /sbin/init 
3749 /init 
3753 /init subcontext
3754 /init subcontext
24801 /usr/bin/xinit /usr/local/bin/croutonxinitrc-wrapper
(bull)z@localhost:~$

**Reply by darkwater4213** [reply]
Now I have to ask...
Are you running Chrome _inside_ the chroot?
bc that's kind of what it looks like lol

That being said, I'm not really sure what you did with awk, so I'm not really sure what's going on.
If you're trying to kill init, I... don't think that's a great idea.
`awk '\$4 ~ /chrome\$/ { print \$3 }'` works, though.

**Comment by TomTravis** [comment]
lmaf no you can see gobal proc via pts mounts in chroot 
no _nohup init to read a new config 
you can awk what you like
the param are more the $10 $15 or $20 onsome tasks

**Reply by darkwater4213** [reply]
Then at the very possible least...
Quit talking to everyone as if they're all idiots.
Not cool.

I was asking an earnest question and you turn around an start spewing gall every which way. Ask yourself: re you here to answer people's questions -- or to lord over people that you have more answers than they do? Think about that very long and very hard. I think you'll find that your actions are somewhat inconsistent with what you want. Certainly with what you ought to want.

Assume the worst in events -- but not in people.

**Reply by TomTravis** [reply]
> Then at the very possible least... Quit talking to everyone as if they're all idiots. Not cool.
> 
> I was asking an earnest question and you turn around an start spewing gall every which way. Ask yourself: re you here to answer people's questions -- or to lord over people that you have more answers than they do? Think about that very long and very hard. I think you'll find that your actions are somewhat inconsistent with what you want. Certainly with what you ought to want.
> 
> Assume the worst in events -- but not in people.

hey did you sign the google CLA i did
your have to be a terrorist to make motion
i used grep -lr ... dirpath| xarg sed -i s|,,,|...|/g alot
systemD is Moving Forward init maybe used to start systemd as chroot does not boot
stop with the hurt feeling you not used to making changes fine

**Reply by dnschneid** [reply]
@TomTravis cut it out; insults aren't helpful or welcome here.

**Comment by TomTravis** [comment]
what is the insult please ?
i moving on no question about this
you behavior is insult also
ask nicely and let it fail leaving people to submit is broken when it is not

**Comment by Dozer257** [comment]
Jesus christ guys calm down, someone here already answered my question a long time ago and I don't even use Crouton or Linux anymore

**Reply by TomTravis** [reply]
it okay it installs Test2 again have a good holiday
